File: 백준/Silver/11660. 구간 합 구하기 5/구간 합 구하기 5.swift
if let input: String = readLine() {
    let arrayOfInput: [Int] = input.split(separator: " ").compactMap { Int($0) }
    let N: Int = arrayOfInput[0]
    let M: Int = arrayOfInput[1]

    var pSum = [[Int]](
        repeating: [Int](
            repeating: 0,
            count: N+1
        ),
        count: N+1
    )

    for y in 1...N {
        if let line: String = readLine() {
            let lines: [Int] = line.split(separator: " ").compactMap { Int($0) }

            for x in 1...N {
                pSum[y][x] = lines[x-1] + pSum[y-1][x] + pSum[y][x-1] - pSum[y-1][x-1]
            }
        }
    }

    var results = [Int]()
    for _ in 0..<M {
        let p: [Int] = readLine()!.split(separator: " ").compactMap { Int($0) }
        let (y1, x1, y2, x2): (Int, Int, Int, Int) = (p[0], p[1], p[2], p[3])
        results.append ( pSum[y2][x2] - pSum[y2][x1-1] - pSum[y1-1][x2] + pSum[y1-1][x1-1] )
    }
    
    print(results.map { String($0) }.joined(separator: "\n") )
}

File: 백준/Silver/11660. 구간 합 구하기 5/구간 합 구하기 5.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

N, M = map(int, input().split())

pSum = [[0 for _ in range(N+1)] for _ in range(N+1)]

for y in range(1, N+1):
    lines = list(map(int, input().split()))

    for x in range(1, N+1):
        pSum[y][x] = lines[x-1] + pSum[y-1][x] + pSum[y][x-1] - pSum[y-1][x-1]
        
results = []
for _ in range(M):
    y1, x1, y2, x2 = map(int, input().split())

    results.append ( pSum[y2][x2] - pSum[y2][x1- 1] - pSum[y1-1][x2] + pSum[y1-1][x1-1] )

print("\n".join(map(str, results))+"\n")